agent/php_api_internal.c

```diff
@@ -103,6 +103,27 @@ PHP_FUNCTION(newrelic_get_hostname) {
   nr_free(hostname);
 }
 
+PHP_FUNCTION(newrelic_get_agent_version) {
+  char* version = NULL;
+
+  NR_UNUSED_HT;
+  NR_UNUSED_RETURN_VALUE_PTR;
+  NR_UNUSED_RETURN_VALUE_USED;
+  NR_UNUSED_THIS_PTR;
+  NR_UNUSED_TSRMLS;
+
+  if (FAILURE == zend_parse_parameters_none()) {
+    RETURN_FALSE;
+  }
+
+#ifdef NR_VERSION
+  version = NR_VERSION;
+#else
+  version = "unknown";
+#endif
```

Comment:
When would `NR_VERSION` not be defined? Why is this conditional construct needed?

Reply:
The check is a guard against an unexpected situation where it is undefined.  Perhaps the compilation should just fail in that scenario.

Reply:
Fixed in commit [7ea6c93](https://github.com/newrelic/newrelic-php-agent/pull/947/commits/7ea6c93be430eeadd4671cc76923a6b17e78c5f8).